### PR TITLE
add verticalKeys to AutocompleteResult

### PR DESCRIPTION
**Pull request by oshi97**
The work has not been done in the backend yet, so only automated testing could be done.

J=SLAP-1708
TEST=auto

### CODE REVIEW

File: tests/infra/AutocompleteServiceImpl.ts [hunk 1]
```diff
@@ -33,6 +34,18 @@ describe('AutocompleteService', () => {
 
   const mockHttpService = new HttpServiceMock();
   const apiResponseValidator = new ApiResponseValidator();
+  function createMockAutocompleteService(params?: { response?: unknown, answersConfig?: AnswersConfig }) {
```
Comment (yen-tt) [inline tests/infra/AutocompleteServiceImpl.ts:37]: can the type be some union of the vertical and universal autocomplete response rather than unknown?

Comment (oshi97) [reply tests/infra/AutocompleteServiceImpl.ts:37]: oh hmm this is the raw api response so I don't think so. I considered making it
```ts
mockAutocompleteResponse | mockAutocompleteResponseWithSections | //etc
```
but didn't really see much value in that

Comment (yen-tt) [reply tests/infra/AutocompleteServiceImpl.ts:37]: oh I see, so can it be of type ApiResponse?

Comment (oshi97) [reply tests/infra/AutocompleteServiceImpl.ts]: ah yeah that's better!

**Review by yen-tt — APPROVED** [review]

**Pull request merged**